### PR TITLE
Send template description to GitHub

### LINKED ISSUE #2477: TechDocs: required description field for docs-template
## Expected Behavior

The fields that are required should be validated before it's possible to create a component.

## Current Behavior

You are able to go through the steps to create a new component but in the generating step it hangs and throws this error.

DEBUG cookiecutter.generate: Check component-info.yaml to see if it's a binary
Unable to create file 'component-info.yaml'
Error message: 'collections.OrderedDict object' has no attribute 'description'
Context: {
    "cookiecutter": {
        "_template": "/template", 
        "component_id": "Nylas", 
        "isOrg": "True", 
        "owner": "tjperry07", 
        "storePath": "tjperry07/backstage-doc-template"
    }
}

### PR DESCRIPTION
**Pull request by dtuite**
This should help people on GitHub understand what the repo is for.

If the user doesn't enter a description then there is no effect on GitHub. The description there is just empty. I believe only the CRA template will run successfully with no description set so that's really the only case where that can happen.

### CODE REVIEW

File: plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts [hunk 1]
```diff
@@ -44,13 +44,14 @@ export class GithubPublisher implements PublisherBase {
     values: RequiredTemplateValues & Record<string, JsonValue>,
   ) {
     const [owner, name] = values.storePath.split('/');
+    const description = String(values.description);
```
Comment (dtuite) [inline plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts:47]: I'm a typeed language beginner so I would appreciate some guidance on what to do here. `description` is a `JsonValue` but `createInOrg` expects it to be `undefined | string`.

Surely there is a better solution than casting it like this? I mean, if I was sending 20 different to GitHub from the `values` would I have to cast all of them?

File: plugins/scaffolder-backend/sample-templates/docs-template/template.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ spec:
   schema:
     required: 
       - component_id
+      - description
```
Comment (emmaindal) [inline plugins/scaffolder-backend/sample-templates/docs-template/template.yaml:20]: Great, thanks @dtuite 💯

File: plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts (hunk 1 above)
Comment (benjdlambert) [reply plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts:47]: You should be able to do `values.description as string`

Comment (benjdlambert) [reply plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts:47]: Or in the `createInOrg` call you can do `description: description ?? ''` but the first one is better.

Comment (dtuite) [reply plugins/scaffolder-backend/src/scaffolder/stages/publish/github.ts]: Thanks Ben. That looks more like Typescript to me.

**Review by Rugvip — APPROVED** [review]
Yep, makes sense 👌

**Comment by timja** [comment]
conflicts ^^, everyone changing that github test atm 👅

**Comment by dtuite** [comment]
Dammit. Everyone stop coding for a minute please. 😂

**Comment by freben** [comment]
image

**Pull request merged**